translator/metric/signalfx/opencensus_to_signalfx.go

```
@@ -0,0 +1,81 @@
+// Copyright 2019, Omnition Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signalfx
+
+import (
+	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	ptypes "github.com/gogo/protobuf/types"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/signalfx/golib/datapoint"
+)
+
+// OCProtoToSignalFx converts a OpenCensus metrics to SignalFx Datapoints.
+func OCProtoToSignalFx(md consumerdata.MetricsData) ([]*datapoint.Datapoint, error) {
+	// Slice with minimum required capacity
+	dps := make([]*datapoint.Datapoint, 0, len(md.Metrics))
+	for _, metric := range md.Metrics {
+		dps = translateAndAppend(dps, md.Resource, md.Node, metric)
+	}
+	return dps, nil
+}
+
+func translateAndAppend(dps []*datapoint.Datapoint, resource *resourcepb.Resource, node *commonpb.Node, metric *metricspb.Metric) []*datapoint.Datapoint {
+	for _, ts := range metric.Timeseries {
+		for _, pt := range ts.Points {
+			dp := &datapoint.Datapoint{}
+			descriptor := metric.GetMetricDescriptor()
+			if descriptor == nil {
+				continue
```

Q.: Should this be recorded (log/metric) somehow?

---

I think it should return an error as this should never ever happen but is still technically possible. I'm going to overhaul the whole exporter and make it production ready each iteration. For now just did the minimal to enable exporting the data that monitors generate.

---

I'd suggest at least to log it, so that it is exposed if it happens. You can handle more robustly in the future.